doc/manual/src/contributing/hacking.md

```diff
@@ -86,6 +86,16 @@ by:
 $ nix develop
 ```
 
+## Pre-commit and formatting
+
+`nix develop` installs Nix-[integrated](https://github.com/cachix/pre-commit-hooks.nix) [pre-commit](https://pre-commit.com) hooks.
```

Comment:
This should be turned off by default. git hooks can be incredible annoying when they are not fully working and you are in the middle of a more complicated git action like a rebase or you are doing a commit which is broken on purpose and will be later fixed. Also when switching to an old branch when still inside the nix develop shell, this will probably blow up.

Reply:
I think it's OK. It's easy enough to use `git commit -n` to skip the Git hooks when needed.

It could be nice to have an environment variable to configure this behavior, like `NIX_DEVELOP_PRE_COMMIT`.